lib/core/src/server/dev-server.ts

```diff
@@ -414,7 +420,7 @@ export async function storybookDevServer(options: any) {
   ]);
 
   // TODO #13083 Remove this when compiling the preview is fast enough
-  if (!options.ci) openInBrowser(address);
+  if (!options.ci) openInBrowser(networkAddress);
```

Comment:
@yannbf - I think it's the case now that we'll want the browser to open to `networkAddress`, given that's what the user is configuring with the `--host` option